lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -3477,7 +3477,7 @@ data WalletException
     | ExceptionConstructSharedWallet ErrConstructSharedWallet
     | ExceptionReadAccountPublicKey ErrReadAccountPublicKey
     | ExceptionSignPayment ErrSignPayment
-    | ExceptionBalanceTx ErrBalanceTx
+    | forall era. ExceptionBalanceTx (ErrBalanceTx era)
```

Comment:
I'm guessing we'll need an `IsRecentEra` constraint here when the `era` actually starts being used.